### PR TITLE
Add support for metric post-proc functions

### PR DESCRIPTION
**Pull request by tjerman**
This allows us to perform simpler operations on end data, such as creating cumulative charts.

REF: https://github.com/cortezaproject/corteza-webapp-compose/pull/182

### CODE REVIEW

File: src/compose/types/chart/index.ts [hunk 1]
```diff
@@ -135,13 +137,24 @@ export class Chart {
     return true
   }
 
-  prepData ({ labels, metrics }: { labels?: Array<string>; metrics?: Array<Array<number>> } = {}, base: any = {}) {
+  prepData ({ labels, metrics }: { labels?: Array<string>; metrics?: Array<Array<number>> } = {}, base: any = {}, postProcFx: (n: number, m: undefined|number, ds: any) => number) {
```
Comment (darh) [inline src/compose/types/chart/index.ts:140]: this interface just got a bit too fat... might be smart if you extract it?

File: src/compose/types/chart/index.ts [hunk 2]
```diff
@@ -135,13 +137,24 @@ export class Chart {
...
     if (labels) {
       base.labels = labels
     }
 
     metrics?.forEach((metric, index) => {
       const ds = base.datasets[index]
+
+      // Dataset post process function
```
Comment (darh) [inline src/compose/types/chart/index.ts:148]: looks like magic, a bit more verbose comment?

File: src/compose/types/chart/index.ts [hunk 3]
```diff
@@ -41,6 +41,8 @@ const defConfig = () => Object.assign({}, {
   },
 })
 
+const defaultFx = 'n'
```
Comment (darh) [inline src/compose/types/chart/index.ts:44]: n?

**Review by darh — CHANGES_REQUESTED** [review]

**Comment by darh** [comment]
where are we with this?

**Comment by tjerman** [comment]
Included with https://github.com/cortezaproject/corteza-js/pull/13

**Pull request closed without merging**